shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -100,21 +192,24 @@ private void load(AlluxioURI filePath, int replication)
         load(newPath, replication);
       }
     } else {
-      Thread thread = CommonUtils.createProgressThread(System.out);
-      thread.start();
-      try {
-        JobGrpcClientUtils.run(new LoadConfig(filePath.getPath(), replication), 3,
-            mFsContext.getPathConf(filePath));
-      } finally {
-        thread.interrupt();
+      if (status.getInAlluxioPercentage() == 100) {
+        // The file has already been fully loaded into Alluxio.
+        System.out.println(filePath + " already in Alluxio fully");
+        return;
+      }
+      if (mFutures.size() >= mThreads) {
+        // Wait one job to complete.
+        waitJob();
```

Comment:
Is there an invariant we are trying to achieve here? There is no guarantee on `mFuture.size` after this code block.

Reply:
mFuture.size is an invariant  when this code block in waitJob(). If one future taked from mDistributedLoadService, it will be deleted from mFuture.

Reply:
You're right, I didn't realize only the RPC call is being parallelized.